app/templates/ui/app/create/create.controller.js

```diff
@@ -4,13 +4,47 @@
   angular.module('app.create')
     .controller('CreateCtrl', CreateCtrl);
 
-  CreateCtrl.$inject = ['$scope', 'MLRest', '$state', 'userService', 'ngToast'];
+  CreateCtrl.$inject = ['$scope', 'MLRest', '$state', 'userService', 'ngToast','x2js','doc'];
 
-  function CreateCtrl($scope, mlRest, $state, userService, toast) {
+  function CreateCtrl($scope, mlRest, $state, userService, toast, x2js, doc) {
     var ctrl = this;
 
-    angular.extend(ctrl, {
-      person: {
+    ctrl.x2js = x2js;
+
+    ctrl.mode = 'create';
+
+    if (doc) {
+      ctrl.mode = 'edit';
+      //check extension
+      if (doc.config.params.format == 'json') {
+        ctrl.person = doc.data;
+      } else {
+        //expect xml format
+        var wrapped = x2js.xml_str2json(doc.data);
+        ctrl.person = wrapped.xml;
+      }
+
+      if (!ctrl.person.tags) {
+        ctrl.person.tags = [];
+      }
+
+      if (!ctrl.person.friends) {
+        ctrl.person.friends = [];
+      }
+
+      if (!ctrl.person.location) {
+        ctrl.person.location = {
+          latitude: 0,
+          longitude: 0
+        };
+      }
+
+      ctrl.uri = doc.config.params.uri;
```

Comment:
doc.config.params.uri? Is that part of the content in the database like that? Isn't it easier to grab uri from `$stateParams`? (like in detail.ctrl)